scipy/io/_fortran.py

```diff
@@ -282,8 +282,8 @@ def read_record(self, *dtypes, **kwargs):
 
         second_size = self._read_size()
         if first_size != second_size:
-            raise IOError('Sizes do not agree in the header and footer for '
-                          'this record - check header dtype')
+            raise ValueError('Sizes do not agree in the header and footer for '
+                             'this record - check header dtype')
```

Comment:
This exception type look more like ValueError, as the potential issue seems to be with the header type. Similar ValueError exceptions are also raised in this method.